storage/database.go

```diff
@@ -518,6 +518,25 @@ func (d *db) FetchBlocksMetadata(
 		pageToken, opts)
 }
 
+func (d *db) FetchBlocksMetadataV2(
+	ctx context.Context,
+	namespace ts.ID,
+	shardID uint32,
+	start, end time.Time,
+	limit int64,
+	pageToken PageToken,
```

Comment:
`PageToken` is just an alias for `[]byte` here.  I wanted to make page tokens opaque to the RPC layer because I need to inspect and form the response way deeper in the call stack (in the shard itself) where there is a much higher level of context about implementation details.